fluxc/src/main/java/org/wordpress/android/fluxc/utils/CommentErrorUtils.java

```diff
@@ -51,24 +57,41 @@ public static FetchedCommentLikesResponsePayload commentErrorToFetchedCommentLik
         return payload;
     }
 
-    public static RemoteCommentResponsePayload commentErrorToPushCommentPayload(BaseNetworkError error,
-                                                                                CommentModel comment) {
+    @NonNull
+    public static RemoteCommentResponsePayload commentErrorToPushCommentPayload(
+            @NonNull BaseNetworkError error,
+            @NonNull CommentModel comment) {
         RemoteCommentResponsePayload payload = new RemoteCommentResponsePayload(comment);
         payload.error = new CommentError(genericToCommentError(error), getErrorMessage(error));
         return payload;
     }
 
-    public static CommentError networkToCommentError(BaseNetworkError error) {
+    @NonNull
+    public static CommentError networkToCommentError(@NonNull BaseNetworkError error) {
         return new CommentError(genericToCommentError(error), getErrorMessage(error));
     }
 
-    private static CommentErrorType genericToCommentError(BaseNetworkError error) {
+    @NonNull
+    private static CommentErrorType genericToCommentError(@NonNull BaseNetworkError error) {
         CommentErrorType errorType = CommentErrorType.GENERIC_ERROR;
         if (error.isGeneric()) {
             switch (error.type) {
                 case INVALID_RESPONSE:
                     errorType = CommentErrorType.INVALID_RESPONSE;
                     break;
+                case TIMEOUT:
+                case NO_CONNECTION:
```

Comment:
Minor, but it looks like a good candidate for simple `if`

```suggestion
        if (error.isGeneric() && error.type == INVALID_RESPONSE) {
                    errorType = CommentErrorType.INVALID_RESPONSE;
```

Reply:
Thanks for noticing that @wzieba , done here: 1b43bf07c2744abac6a23b324e613fe8df607b16

PS: I didn't apply your commit suggestion as I noticed that the `ommentErrorType.` were missing on the if statement. 😊